code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,344 @@
+			boulders_contained -= potential_boulder
+			break
+		if(boulders_concurrent <= 0)
+			break //Try again next time
+
+		if(!istype(potential_boulder, /obj/item/boulder))
+			potential_boulder.forceMove(drop_location())
+			CRASH("\The [src] had a non-boulder in it's boulders contained!")
+
+		var/obj/item/boulder/boulder = potential_boulder
+		if(!check_for_processable_materials(boulder.custom_materials)) //Checks for any new materials we can process.
+			boulders_concurrent-- //We count skipped boulders
+			remove_boulder(boulder)
+			continue
+		boulders_concurrent--
+		boulder.durability-- //One less durability to the processed boulder.
+		if(COOLDOWN_FINISHED(src, sound_cooldown))
+			COOLDOWN_START(src, sound_cooldown, 1.5 SECONDS)
+			playsound(loc, usage_sound, 40, FALSE, SHORT_RANGE_SOUND_EXTRARANGE) //This can get annoying. One play per process() call.
+		stop_processing_check = TRUE
+		if(boulder.durability <= 0)
+			breakdown_boulder(boulder) //Crack that bouwlder open!
+			continue
+	if(!stop_processing_check)
+		playsound(src.loc, 'sound/machines/ping.ogg', 50, FALSE)
+		return PROCESS_KILL
+
+
+/obj/machinery/bouldertech/CanAllowThrough(atom/movable/mover, border_dir)
+	if(!anchored)
+		return FALSE
+	if(boulders_contained.len >= boulders_held_max)
+		return FALSE
+	if(istype(mover, /obj/item/boulder))
+		var/obj/item/boulder/boulder = mover
+		return boulder.can_get_processed()
+	return ..()
+
+/obj/machinery/bouldertech/examine(mob/user)
+	. = ..()
+	if(holds_mining_points)
+		. += span_notice("The machine reads that it has [span_bold("[points_held] mining points")] stored. Swipe an ID to claim them.")
+
+/**
+ * Accepts a boulder into the machinery, then converts it into minerals.
+ * If the boulder can be fully processed by this machine, we take the materials, insert it into the silo, and destroy the boulder.
+ * If the boulder has materials left, we make a copy of the boulder to hold the processable materials, take the processable parts, and eject the original boulder.
+ * @param chosen_boulder The boulder to being breaking down into minerals.
+ */
+/obj/machinery/bouldertech/proc/breakdown_boulder(obj/item/boulder/chosen_boulder)
+	if(QDELETED(chosen_boulder))
+		return FALSE
+	if(chosen_boulder.loc != src)
+		return FALSE
+	if(!length(chosen_boulder.custom_materials))
+		qdel(chosen_boulder)
+		playsound(loc, 'sound/weapons/drill.ogg', 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		update_boulder_count()
+		return FALSE
+	if(isnull(silo_materials))
+		return FALSE
+
+	//here we loop through the boulder's ores
+	var/list/processable_ores = list()
+	var/tripped = FALSE
+	//If a material is in the boulder's custom_materials, but not in the processable_materials list, we add it to the processable_ores list to add back to a leftover boulder.
+	for(var/datum/material/possible_mat as anything in chosen_boulder.custom_materials)
+		if(!is_type_in_list(possible_mat, processable_materials))
+			continue
```

Comment:
This meant to be an early `continue` right? Because all lines after this in the same indendation as `continue` so all of them get skipped. You should move all statements after this outside